### ISSUE TITLE
Running test suite as a whole fails

### ISSUE THREAD

**Issue by SeanFeldman** [Azure/azure-service-bus-dotnet#17]
Using [latest `master` branch](f2bfa77bd0c38910ddf284305ffb515d534b44ae) I am not able to run the entire test suite.

XUnit test runner result:

![image](https://cloud.githubusercontent.com/assets/1309622/20785687/bef07208-b75e-11e6-949d-822ff396ad5b.png)

Peeked messages:

![image](https://cloud.githubusercontent.com/assets/1309622/20785711/dcecf5b0-b75e-11e6-94e8-7edf04ecabe4.png)

I've modified the label of the sent messages to see what tests are causing messages to remain on the queue and see that those are the deferred messages that never get cleaned up:

![image](https://cloud.githubusercontent.com/assets/1309622/20785945/40ca3f56-b760-11e6-943d-451160387b1e.png)

Also couldn't receive and delete all the messages. Some are stack in the queue and have to remove the queue.

**Questions**

1. Is `MessageId` `nil` a normal thing? I was pretty sure `MessageId` should be always set to a value. Please educate me. (Separate issue #19)
1. Should tests always run with a clean slate and not leave "garbage" behind? With deferred messages it's not possible, they will be piling up. (potentially a separate issue /cc @jtaubensee)
1. Executing individual tests results should not be different from executing the entire suite. Once build server is in place, this will start hindering the work flow. Should we address it sooner than later? (potentially a separate issue /cc @jtaubensee)

**Comment by jtaubensee** [comment]
PR's #47, #58, #59, should now provide us with an automated build. Once this builds successfully, we can close this issue. I think to your point @SeanFeldman, there are a couple of tests that are throwing errors. Will update once the build finishes.

**Comment by jtaubensee** [comment]
Here are the 4 failing AppVeyor tests:

https://ci.appveyor.com/project/jtaubensee/azure-service-bus-dotnet/build/1.0.28

* `TopicClientPeekLockWithAbandonTestCase(topicName: "partitioned-topic")`
* `TopicClientReceiveDeleteTestCase(topicName: "partitioned-topic")`
* `TopicClientPeekLockDeferTestCase(topicName: "partitioned-topic")`
* `TopicClientRenewLockTestCase(topicName: "partitioned-topic")`

cc - @vinaysurya

**Comment by jtaubensee** [comment]
Fixed by #61
